validator/client/validator.go

```diff
@@ -259,25 +259,28 @@ func (v *validator) UpdateAssignments(ctx context.Context, slot uint64) error {
 // RolesAt slot returns the validator roles at the given slot. Returns nil if the
 // validator is known to not have a roles at the at slot. Returns UNKNOWN if the
 // validator assignments are unknown. Otherwise returns a valid ValidatorRole map.
-func (v *validator) RolesAt(slot uint64) map[[48]byte]pb.ValidatorRole {
-	rolesAt := make(map[[48]byte]pb.ValidatorRole)
+func (v *validator) RolesAt(slot uint64) map[[48]byte][]pb.ValidatorRole {
+	rolesAt := make(map[[48]byte][]pb.ValidatorRole)
 	for _, assignment := range v.assignments.ValidatorAssignment {
-		var role pb.ValidatorRole
-		switch {
-		case assignment == nil:
-			role = pb.ValidatorRole_UNKNOWN
-		case assignment.ProposerSlot == slot && assignment.AttesterSlot == slot:
-			role = pb.ValidatorRole_BOTH
-		case assignment.AttesterSlot == slot:
-			role = pb.ValidatorRole_ATTESTER
-		case assignment.ProposerSlot == slot:
-			role = pb.ValidatorRole_PROPOSER
-		default:
-			role = pb.ValidatorRole_UNKNOWN
+		var roles []pb.ValidatorRole
+
+		if assignment == nil {
+			roles = append(roles, pb.ValidatorRole_UNKNOWN)
```

Comment:
I think we can just leave it empty? No need to insert UNKNOWN when an empty roles array means no work to do